### DISCUSSION TITLE
`gpt-4-turbo` inquiry

### DISCUSSION THREAD

**Discussion by luke3butler** [danny-avila/LibreChat#2504] [Q&A] [answered]
### What happened?

When specifying the `gpt-4-turbo` model, it [appears to be using `gpt-4-1106`](https://github.com/danny-avila/LibreChat/blob/3df4fac118179929968e6c7684f94e9699b8f55b/api/models/tx.js#L62C13-L62C23).

The [OpenAI definition](https://platform.openai.com/docs/models/gpt-4-turbo-and-gpt-4) of `gpt-4-turbo` is `gpt-4-turbo-2024-04-09`.

### Steps to Reproduce

`OPENAI_MODELS=gpt-4-turbo`

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Except in very unique cases (multimodal requests, azure), LibreChat does not replace the model name and sends exactly what you select to the API.

When I select `gpt-4-turbo`, the model field in the request body is `gpt-4-turbo`:

![image](https://github.com/danny-avila/LibreChat/assets/110412045/dd8a5875-e33a-4bcc-9a50-bc56a5415ff4)

![image](https://github.com/danny-avila/LibreChat/assets/110412045/4923223c-f659-4fdc-bf0b-f7f3271c7ee5)

If OpenAI is treating `gpt-4-turbo` differently from `gpt-4-turbo-2024-04-09`, just use this value in your models list instead

![image](https://github.com/danny-avila/LibreChat/assets/110412045/ee605d0d-82c2-42aa-a195-1b9163c851bf)

![image](https://github.com/danny-avila/LibreChat/assets/110412045/572a9a35-0545-4f9c-b92d-5b1555887217)

**Comment by danny-avila** [comment, marked as answer]
> When specifying the `gpt-4-turbo` model, it [appears to be using `gpt-4-1106`](https://github.com/danny-avila/LibreChat/blob/3df4fac118179929968e6c7684f94e9699b8f55b/api/models/tx.js#L62C13-L62C23).

This is simply for identifying the context length, and the output of this function is exclusively for default context management.
